src/test/java/io/confluent/examples/streams/interactivequeries/WordCountInteractiveQueriesExampleTest.java

```diff
@@ -69,21 +69,40 @@
       "interactive-queries-wordcount-example-word-count-changelog";
   private static final String WINDOWED_WORD_COUNT_OUTPUT =
       "interactive-queries-wordcount-example-windowed-word-count-changelog";
+  
+  private static final List<String> inputValues = Arrays.asList("hello",
+          "world",
+          "world",
+          "hello world",
+          "all streams lead to kafka",
+          "streams",
+          "kafka streams");
 
   @Rule
   public final TemporaryFolder temp = new TemporaryFolder();
   private KafkaStreams kafkaStreams;
   private WordCountInteractiveQueriesRestService proxy;
 
   @BeforeClass
-  public static void createTopic() {
+  public static void createTopicsAndProduceDataToInputTopics() throws Exception {
     CLUSTER.createTopic(WordCountInteractiveQueriesExample.TEXT_LINES_TOPIC, 2, 1);
     // The next two topics don't need to be created as they would be auto-created
     // by Kafka Streams, but it just makes the test more reliable if they already exist
     // as creating the topics causes a rebalance which closes the stores etc. So it makes
     // the timing quite difficult...
     CLUSTER.createTopic(WORD_COUNT, 2, 1);
     CLUSTER.createTopic(WINDOWED_WORD_COUNT, 2, 1);
+  
+    // Produce sample data to the input topic before the tests starts.
+    Properties producerConfig = new Properties();
+    producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, CLUSTER.bootstrapServers());
+    producerConfig.put(ProducerConfig.ACKS_CONFIG, "all");
+    producerConfig.put(ProducerConfig.RETRIES_CONFIG, 0);
+    producerConfig.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+    producerConfig.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class);
+    IntegrationTestUtils
+            .produceValuesSynchronously(WordCountInteractiveQueriesExample.TEXT_LINES_TOPIC, inputValues,
+                    producerConfig);
```

Comment:
I think it would be better to keep this in the method(s) that need(s) it. The reason is that not all tests needs to consume this data, but it will get populated before every test. It's logically fine, but it does cause our tests to run slower.

In fact, I'd recommend leaving this method as you have it, and just removing the `@BeforeClass` annotation and directly calling this method at the start of the relevant tests.

Reply:
@vvcephei, Maybe there's some confusion, Actually, The @BeforeClass annotation ensures that any *computationally expensive setup is executed only once* for all those tests that share this setup.
However, @Before will run before each test case.

Reference: http://junit.sourceforge.net/javadoc/org/junit/BeforeClass.html

Reply:
Ah, of course. I misread the code. My apologies.